scripts/system_performance/system_performance_queries.py

```diff
@@ -0,0 +1,162 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+
+"""
+The purpose of this script is for performing MySQL queries to monitor system state
+performance and health.
+"""
+from __future__ import print_function
+import itertools
+import logging
+from utils.clients.database_client import DatabaseClient
+from utils.clients.connection_exception import ConnectionException
+
+
+class DatabaseMonitorChecks(object):
+    """Class to query system performance"""
+    table = "reduction_viewer_reductionrun"
+
+    # Establishing a connection with Database using Database Client
+    def __init__(self):
+        self.database = DatabaseClient()
+        try:
+            self.connection = self.database.connect()
+        except ConnectionException:
+            raise ConnectionException('database')
+
+    def query_log_and_execute(self, constructed_query):
+        """Logs and executes all queries ran in script"""
+        logging.info('SQL QUERY: %s', constructed_query)
+        print(constructed_query)
+        return self.connection.execute(constructed_query).fetchall()
+
+    def instruments_list(self):
+        """Retrieve current list of instruments"""
+        all_instruments = "SELECT id, name "\
+                          "FROM reduction_viewer_instrument"
+        return self.query_log_and_execute(all_instruments)
+
+    def rb_range_by_instrument(self, instrument, start_date, end_date):
+        """Retrieves run_number column and return missing sequential values """
+        missing_rb_calc_vars = {}
+
+        missing_rb_query = "SELECT run_number "\
+                           "FROM {} " \
+                           "WHERE instrument_id = {} " \
+                           "AND created " \
+                           "BETWEEN '{}' " \
+                           "AND '{}'".format(DatabaseMonitorChecks.table,
+                                             instrument,
+                                             start_date,
+                                             end_date)
+        missing_rb_calc_vars['run_numbers'] = self.query_log_and_execute(missing_rb_query)
+        # pylint: disable=line-too-long
+        # Converts list of run number sets containing longs into list of integers [(123L), (456L)] -> [123, 456]
+        return [int(elem) for elem in list(itertools.chain.from_iterable(missing_rb_calc_vars['run_numbers']))]
+        # pylint: enable=line-too-long
+
+    @staticmethod
+    def query_sub_segment_replace(query_arguments):
```

Comment:
Could probably improve name of this function too. I think it's a bit vague compared to the functionality actually happening here

Reply:
specifically these all refer to time period / scale perhaps there is a name around this. If we can't think of a name for this, perhaps we'll have to to have a look at splitting up this function

Reply:
would `time_scale_format_segment_replace` be an appropriate method name?